webapp/components/create_post.jsx

```diff
@@ -134,6 +166,18 @@ export default class CreatePost extends React.Component {
 
         this.setState({submitting: true, serverError: null});
 
+        const stats = ChannelStore.getCurrentStats();
+        const members = stats.member_count - 1;
+
+        if ((post.message.includes('@all') || post.message.includes('@channel')) && members >= 5) {
+            this.setState({
+                showConfirmModal: true,
+                postMessage: post,
```

Comment:
I don't really like storing the post separately from the other state that represents its fields since there might be some confusion later on about the difference between `this.state.message` and `this.state.postMessage`. If you make the changes I suggested above, you can have the new function construct `post` itself instead of storing it in the state